os/board/rtl8721csm/src/component/os/tizenrt/tizenrt_service.c

```diff
@@ -681,30 +704,87 @@ static int _tizenrt_create_task(struct task_struct *ptask, const char *name, u32
 	ptask->task = NULL;
 	ptask->task_name = NULL;
 	return _FAIL;
+#else
+	int func_addr, ctx_addr;
+	pid_t pid;
+	char str_func_addr[8+1];
+	char str_ctx_addr[8+1];
+	char *task_info[2 + 1];
+	priority = SCHED_PRIORITY_DEFAULT + priority;
+	priority = (priority > SCHED_PRIORITY_MAX || priority < SCHED_PRIORITY_MIN)?SCHED_PRIORITY_DEFAULT:priority;
+	stack_size *= sizeof(uint32_t);
+	/* Execute loading thread for load all binaries */
+	func_addr = (int)func;
+	ctx_addr = (int)thctx;
+	task_info[0] = itoa(func_addr, str_func_addr, 16);
+	task_info[1] = itoa(ctx_addr, str_ctx_addr, 16);
+	task_info[2] = NULL;
+	pid = kernel_thread(name, priority, stack_size, wrapper_thread, (char * const *)task_info);
+	if (pid == ERROR) {
+		DBG_ERR("%s fail\n", name);
+		return _FAIL;
+	}
+	ptask->task = pid;
+	ptask->task_name = name;
+	return _SUCCESS;
+#endif
 }
 
 static void _tizenrt_delete_task(struct task_struct *ptask)
 {
+#if USE_PTHREAD_MUTEX
 	int status = 0;
 	pthread_t *tid = (pthread_t *) ptask->task;
 	if (!ptask->task || *tid == 0) {
 		DBG_ERR("_tizenrt_delete_task(): ptask is NULL %s!\n", ptask->task_name);
 		return;
 	}
-	//status = task_delete(ptask->task); //TODO
 	status = pthread_cancel(*tid);
 	if (status != OK) {
 		DBG_ERR("delete the task failed!", ptask->task_name);
 		return;
 	}
 	_tizenrt_mfree(tid, sizeof(*tid));
 	ptask->task = NULL;
+#else
+	pid_t pid;
+	int status;
+	pid = (pid_t) ptask->task;
```

Comment:
Please remove space after type casting.

Reply:
pid is local and it have not been used or declare in this function.